cosmo_tester/test_suites/bootstrap_based_tests/sanity_scenario_test.py

```diff
@@ -0,0 +1,174 @@
+
+    for manager in hosts.instances[1:]:
+        manager.upload_plugins = False
+
+    try:
+        hosts.create()
+        yield hosts.instances
+    finally:
+        hosts.destroy()
+
+
+@pytest.fixture(scope='function')
+def nodecellar(request, cfy, managers, attributes, ssh_key, tmpdir, logger):
+    manager = managers[0]
+    manager.use()
+    tenant = prepare_and_get_test_tenant(TENANT_NAME, manager, cfy)
+    nc = NodeCellarExample(
+        cfy, manager, attributes, ssh_key, logger, tmpdir,
+        tenant=tenant, suffix='simple')
+    nc.blueprint_file = 'simple-blueprint-with-secrets.yaml'
+    yield nc
+
+
+def test_sanity_scenario(managers,
+                         cfy,
+                         logger,
+                         tmpdir,
+                         attributes,
+                         nodecellar):
+    manager1 = managers[0]
+    manager2 = managers[1]
+    manager3 = managers[2]
+
+    logger.info('Cfy version')
+    cfy('--version')
+
+    logger.info('Cfy status')
+    cfy.status()
+
+    # Create user and add the new user to the tenant
+    _configure_tenant(cfy, manager1, logger)
+
+    _set_sanity_user(cfy, manager1, logger)
+
+    # Creating secrets
+    _create_secrets(cfy, logger, manager1)
+
+    nodecellar.upload_and_verify_install()
+
+    _set_admin_user(cfy, manager1, logger)
+
+    _start_cluster(cfy, manager1, logger)
+
+    _set_admin_user(cfy, manager2, logger)
+
+    _join_cluster(cfy, manager1, manager2, logger)
+
+    logger.info('Setting replica manager')
+    # ha_helper.set_active(manager2, cfy, logger)   # bug in agents upgrade
+
+    snapshot_id = 'SNAPSHOT_ID'
+    local_snapshot_path = str(tmpdir / 'snap.zip')
+    logger.info('Creating snapshot')
+    create_snapshot(manager1, snapshot_id, attributes, logger)
+    download_snapshot(manager1, local_snapshot_path, snapshot_id, logger)
+
+    _set_admin_user(cfy, manager3, logger)
+
+    logger.info('Uploading and restoring snapshot')
+    upload_snapshot(manager3, local_snapshot_path, snapshot_id, logger)
+    restore_snapshot(manager3, snapshot_id, cfy, logger)
+    verify_services_status(manager3)
+    # wait a while to allow the restore-snapshot post-workflow commands to run
+    time.sleep(30)
+
+    # upgrade_agents(cfy, manager3, logger)          # bug in agents upgrade
+
+    _set_sanity_user(cfy, manager3, logger)
+
+    # nodecellar.uninstall()                         # bug in agents upgrade
+
+
+def _create_secrets(cfy, logger, manager1):
+    logger.info('Creating secret agent_user as blueprint input')
+    cfy.secrets.create('agent_user', '-s', 'centos')
+
+    logger.info('Creating secret agent_private_key_path as blueprint input')
+    cfy.secrets.create('agent_private_key_path', '-s',
+                       manager1.remote_private_key_path)
+
+    logger.info('Creating secret host_ip as blueprint input')
+    cfy.secrets.create('host_ip', '-s', manager1.ip_address)
+
+
+def _configure_tenant(cfy, manager, logger):
```

Comment:
This function creates a user. Why is it called configure tenant?

Reply:
fixed